zerver/views/invite.py

```diff
@@ -185,6 +186,38 @@ def resend_user_invite_email(
     return json_success(request, data={"timestamp": timestamp})
 
 
+@require_member_or_admin
+@has_request_variables
+def edit_multiuse_invite(
+    request: HttpRequest,
+    user_profile: UserProfile,
+    invite_id: int,
+    stream_ids: List[int] = REQ(json_validator=check_list(check_int)),
+    invite_as: int = REQ(json_validator=check_int),
+) -> HttpResponse:
+    try:
+        invite = MultiuseInvite.objects.get(id=invite_id)
+    except MultiuseInvite.DoesNotExist:
+        raise JsonableError(
+            _("Invite does not exist with id: {invite_id}.").format(invite_id=invite_id)
+        )
+
+    check_role_based_permissions(invite_as, user_profile, require_admin=True)
+
+    streams: List[Stream] = []
+    for stream_id in stream_ids:
+        try:
+            (stream, sub) = access_stream_by_id(user_profile, stream_id)
+        except JsonableError:
+            raise JsonableError(
+                _("Stream does not exist with id: {stream_id}.").format(stream_id=stream_id)
+            )
+        streams.append(stream)
+
+    do_edit_multiuse_invite_link(invite, invite_as, streams)
+    return json_success(request)
```

Comment:
This seems to do only about half the security checks from `generate_multiuse_invite_backend`. The result is that your code likely introduces at least one security bug. Please be careful about this sort of thing.